src/pages/journal/components/test-slot/test-slot.ts

```diff
@@ -54,10 +58,13 @@ export class TestSlotComponent implements SlotComponent, OnInit, OnDestroy {
         select(getTests),
         select(tests => getTestStatus(tests, slotId)),
       ),
+      testActivityCode$: this.store$.pipe(
+        select(getTests),
+        map(tests => getActivityCodeBySlotId(tests, this.slot.slotDetail.slotId)),
+      ),
     };
 
     const { testStatus$ } = this.componentState;
-
     this.subscription = merge(testStatus$).subscribe();
```

Comment:
@roberthall12 We can also get rid of this subscription and therefore the call to unsubscribe too. Realise its not part of your PR but would be nice to keep things clean. Happy with this once that's done so will approve 👍